File: server/libraryBackend/src/main/java/com/jnit/library/entity/Book.java
package com.jnit.library.entity;


import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Date;

@Entity
@Table
@NoArgsConstructor
@AllArgsConstructor
@Data
@JsonIgnoreProperties({"hibernateLazyInitializer"})
public class Book {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String bookName;

    private String author;
    private String takenBy;

    private Date takenDate;

    private Date returnDate;

    @Column(name = "description", columnDefinition = "text")
    private String description;


    private Boolean available;

}

File: client/libraryFrontEnd/src/app/model/Book.ts
export class Book {
  id: number;
  bookName: string;
  author: string;
  takenBy: string;
  takenDate: Date;
  returnDate: Date;
  description: string;
  available: boolean;

  constructor(
    id = 0,
    bookName = '',
    author = '',
    takenBy = '',
    takenDate = new Date(),
    returnDate = new Date(),
    description = '',
    available = false
  ) {
    this.id = id;
    this.bookName = bookName;
    this.author = author;
    this.takenBy = takenBy;
    this.takenDate = takenDate;
    this.returnDate = returnDate;
    this.description = description;
    this.available = available;
  }
}
